src/main/java/com/session/service/client/SessionClientImpl.java

```diff
@@ -74,15 +74,18 @@ public Session getSessionByID(String sessionID) throws SessionClientException {
         ZebedeeSession session = null;
 
         if (StringUtils.isNotEmpty(sessionID)) {
-            HttpGet httpGet = http.createHttpGet(host, "/session/" + sessionID);
-
             try {
-                session = http.doGet(httpGet, getSessionResponseHandler());
+                session = http.get(host, "/sessions/" + sessionID, getSessionResponseHandler());
             } catch (IOException ex) {
-                throw new SessionClientException("TODO", ex);
+                String msg = format("unable to retrieve session: {0}", ex);
+                throw new SessionClientException(msg);
```

Comment:
May we well inline the `format` in the exception constructor.

Reply:
Updated for some of the other messages too.